src/manager/manager.c

```diff
@@ -265,35 +265,49 @@ bool manager_set_port(Manager *manager, const char *port_s) {
 }
 
 bool manager_parse_config(Manager *manager, const char *configfile) {
-        _cleanup_config_ config *config = NULL;
-        topic *topic = NULL;
-        const char *port = NULL;
+        int result = 0;
 
-        config = parsing_ini_file(configfile);
-        if (config == NULL) {
+        result = config_initialize();
+        if (result != 0) {
+                fprintf(stderr, "Error initializing configuration, error code '%d'.", result);
                 return false;
         }
 
-        // set defaults for logging
-        hirte_log_init();
-        // overwrite default log settings with config
-        hirte_log_init_from_config(config);
-        // overwrite config settings with env vars
-        hirte_log_init_from_env();
+        result = config_load_complete_cofiguration(
+                        NULL, // TODO: https://github.com/containers/hirte/issues/147
+                        CFG_ETC_HIRTE_CONF,
+                        NULL); // TODO: https://github.com/containers/hirte/issues/148
+        if (result != 0) {
+                fprintf(stderr, "Error loading configuration '%s', error code '%d'.", CFG_ETC_HIRTE_CONF, result);
+                config_dispose();
+                return false;
+        }
 
-        topic = config_lookup_topic(config, "Manager");
-        if (topic == NULL) {
-                return true;
+        // TODO: Do we really need to support custom config files passed as parameters from command line?
+        if (configfile != NULL) {
+                result = config_load_from_file(configfile);
+                if (result != 0) {
+                        fprintf(stderr,
+                                "Error loading configuration file '%s', error code '%d'.",
+                                configfile,
+                                result);
+                        config_dispose();
```

Comment:
I think `config_dispose()` needs to be called also for the other fail-case returns in this functions (e.g. when `manager_set_port` fails). Instead of this, how about calling `config_dispose()` on the callers side? But then we'd have the same issue in `main.c`. Maybe the config should be part of the manger struct and the `config_dispose` should be called only once when the manager is freed. What do you think?

Reply:
Hmm, not sure where to handle it. It there a way to create some hook reacting on KILL or TERM signal? Registering a function reacting on this signal might be a good way how to free objects, which should live the whole application lifetime

Reply:
Yes, we have the `service/shutdown.c` where we also add callbacks to the signals:
https://github.com/containers/hirte/blob/main/src/libhirte/service/shutdown.c#L88-L132
These signals are setup when starting the manager and ensures that while running a SIGTERM all memory is freed properly. 

So I think the lifetime of this `hashmap` should be linked to the [manager struct](https://github.com/containers/hirte/blob/main/src/manager/manager.h#L10-L31). So in the [manager_new](https://github.com/containers/hirte/blob/main/src/manager/manager.c#L19-L48) where we could allocate it, e.g.
```C
// ...
manager->config = config_new(); // maybe rename from config_initialize and return the hashmap instead of an int
```
and in [manager_unref](https://github.com/containers/hirte/blob/main/src/manager/manager.c#L55-L100) we can call free(hashmap).

Reply:
OK, I've added invocation of `config_dispose()` to both `manager_unref()` and `agent_unref()`